tests/unit/lms/db/_locks_test.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+
+from lms.db import CouldNotAcquireLock, LockType, try_advisory_transaction_lock
+
+# Two very random IDs which shouldn't conflict with any tests that happen to
+# be running in parallel with this module.
+LOCK_ID = 1_000_001
+OTHER_LOCK_ID = 1_000_002
+
+
+# Make sure locks acquired in different tests don't conflict.
+@pytest.mark.xdist_group("TryAdvisoryTransactionLock")
+class TestTryAdvisoryTransactionLock:
+    def test_it_succeeds_if_lock_available(self, db_session):
+        try_advisory_transaction_lock(
+            db_session, LockType.OAUTH2_TOKEN_REFRESH, LOCK_ID
+        )
+
+    def test_it_fails_if_lock_not_available(self, db_session, other_db_session):
+        try_advisory_transaction_lock(
+            db_session, LockType.OAUTH2_TOKEN_REFRESH, LOCK_ID
+        )
```

Comment:
I might consider having the test call the underlying Postgres function directly here, rather than calling the method-under-test as part of the test setup (same applies to a couple of the other tests below)